Symfony/CS/Doctrine/Annotation/Tokens.php

```diff
@@ -0,0 +1,342 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Symfony\CS\Doctrine\Annotation;
+
+use Symfony\CS\DocBlock\Tag;
+
+/**
+ * A list of Doctrine annotation tokens.
+ */
+class Tokens extends \SplFixedArray
+{
+    /**
+     * Returns whether the token at the given index is a Doctrine annotation.
+     *
+     * @param int $index
+     *
+     * @return bool
+     */
+    public function isDoctrineAnnotation($index)
+    {
+        if (!$this[$index]->isType(Token::T_AT)) {
+            return false;
+        }
+
+        if (!$this[$index + 1]->isType(Token::T_IDENTIFIER)) {
+            return false;
+        }
+
+        if (!$this[$index - 1]->isWhitespace() && !$this->isInsideDoctrineAnnotation($index)) {
+            return false;
+        }
+
+        $identifier = $this[$index + 1]->getContent();
+        $tag = new Tag('@'.$identifier);
+        if ($tag->valid()) {
+            return false;
+        }
+
+        $phpUnitAnnotations = array(
```

Comment:
Unfortunately there is no exhaustive list of valid Doctrine annotations because one can create custom annotations, see [Doctrine Annotation documentation](http://docs.doctrine-project.org/projects/doctrine-common/en/latest/reference/annotations.html#annotation-classes).

Maybe this should be an option with the current list as a sensible default value?